packages/nuxt/src/components/plugins/lazy-hydration-macro-transform.ts

```diff
@@ -0,0 +1,115 @@
+import { createUnplugin } from 'unplugin'
+import type { Component, ComponentsOptions } from 'nuxt/schema'
+import { relative } from 'pathe'
+
+import MagicString from 'magic-string'
+import { genDynamicImport, genImport } from 'knitwork'
+import { pascalCase, upperFirst } from 'scule'
+import { isJS, isVue } from '../../core/utils'
+
+interface LoaderOptions {
+  getComponents (): Component[]
+  srcDir: string
+  sourcemap?: boolean
+  transform?: ComponentsOptions['transform']
+  clientDelayedComponentRuntime: string
+}
+
+const LAZY_HYDRATION_MACRO_RE = /(?:\b(?:const|let|var)\s+(\w+)\s*=\s*)?defineLazyHydrationComponent\(\s*['"]([^'"]+)['"]\s*,\s*\(\s*\)\s*=>\s*import\s*\(\s*['"]([^'"]+)['"]\s*\)\s*\)/g
+const COMPONENT_NAME = /import\(["'].*\/([^\\/]+?)\.\w+["']\)/
+const HYDRATION_STRATEGY = ['visible', 'idle', 'interaction', 'mediaQuery', 'if', 'time', 'never']
+
+export const LAZY_HYDRATION_MAGIC_COMMENT = '/* #__LAZY_HYDRATION_MACRO__ */'
```

Comment:
I think we can actually avoid the need for this by importing with a prefix, e.g. `import { createLazyInteractionComponent as _createLazyInteractionComponent }`. Rollup will deduplicate these with any imports added in the component loader. And we avoid the costliness/flakiness of checking for magic comments + parsing AST.